gno.land/pkg/gnoland/app.go

```diff
@@ -20,34 +22,52 @@ import (
 	"github.com/gnolang/gno/tm2/pkg/store/iavl"
 )
 
-// NewApp creates the GnoLand application.
-func NewApp(rootDir string, skipFailingGenesisTxs bool, logger log.Logger, maxCycles int64) (abci.Application, error) {
-	// Get main DB.
-	db, err := dbm.NewDB("gnolang", dbm.GoLevelDBBackend, filepath.Join(rootDir, "data"))
-	if err != nil {
-		return nil, fmt.Errorf("error initializing database %q using path %q: %w", dbm.GoLevelDBBackend, rootDir, err)
+type CustomAppConfig struct {
+	DB                    dbm.DB
+	GnoRootDir            string
+	SkipFailingGenesisTxs bool
+	Logger                log.Logger
+	MaxCycles             int64
+}
+
+func (c *CustomAppConfig) applyDefault() {
+	if c.Logger == nil {
+		c.Logger = log.NewNopLogger()
+	}
+
+	if c.DB == nil {
+		c.DB = dbm.NewMemDB()
 	}
 
+	if c.GnoRootDir == "" {
+		c.GnoRootDir = guessGnoRootDir()
+	}
+}
+
+// NewApp creates the GnoLand application.
+func NewCustomApp(cfg CustomAppConfig) (abci.Application, error) {
+	cfg.applyDefault()
+
 	// Capabilities keys.
 	mainKey := store.NewStoreKey("main")
 	baseKey := store.NewStoreKey("base")
 
 	// Create BaseApp.
-	baseApp := sdk.NewBaseApp("gnoland", logger, db, baseKey, mainKey)
+	baseApp := sdk.NewBaseApp("gnoland", cfg.Logger, cfg.DB, baseKey, mainKey)
 	baseApp.SetAppVersion("dev")
 
 	// Set mounts for BaseApp's MultiStore.
-	baseApp.MountStoreWithDB(mainKey, iavl.StoreConstructor, db)
-	baseApp.MountStoreWithDB(baseKey, dbadapter.StoreConstructor, db)
+	baseApp.MountStoreWithDB(mainKey, iavl.StoreConstructor, cfg.DB)
+	baseApp.MountStoreWithDB(baseKey, dbadapter.StoreConstructor, cfg.DB)
 
 	// Construct keepers.
 	acctKpr := auth.NewAccountKeeper(mainKey, ProtoGnoAccount)
 	bankKpr := bank.NewBankKeeper(acctKpr)
-	stdlibsDir := filepath.Join("..", "gnovm", "stdlibs")
-	vmKpr := vm.NewVMKeeper(baseKey, mainKey, acctKpr, bankKpr, stdlibsDir, maxCycles)
+	stdlibsDir := filepath.Join(cfg.GnoRootDir, "gnovm", "stdlibs")
```

Comment:
be careful with the naming, `cfg.GnoRootDir` is the git clone of the gno repo where we can find the `examples` folder and other folders, while you're also using `rootDir string` below which is the gnoland server root dir, used to create the configuration files.

Should be more explicit.